Disambiguate registration policy definitions (#41)
ietf-scitt/draft-birkholz-scitt-architecture: draft-birkholz-scitt-architecture.md

Policy Name | Required `reg_info` attributes | Implementation
---|---|---

> TimeLimited | `register_by: uint` | Returns true if now () < register_by at registration time. The ledger MUST store the ledger time at registration along with the claim, and SHOULD indicate it in receipts. The value provided for `register_by` MUST be a Unix timestamp e.g. a `uint` representing the number of seconds since 00:00:00 UTC, 1 January 1970, minus leap seconds.

Reviewer: ```suggestion
TimeLimited | `register_by: uint` | Returns true if now () < register_by at registration time. The ledger MUST store the ledger time at registration along with the claim, and SHOULD indicate it in receipts. The value provided for `register_by` MUST be a Unix timestamp, i.e., a `uint` representing the number of seconds since 00:00:00 UTC, 1 January 1970, without leap seconds.
```

> TimeLimited | `register_by: uint` | Returns true if now () < `register_by` at registration time. The ledger MUST store the ledger time at registration along with the claim, and SHOULD indicate it in receipts. The value provided for `register_by` MUST be a Unix timestamp, i.e., a `uint` representing the number of seconds since 00:00:00 UTC, 1 January 1970, without leap seconds.

Reviewer 2: there is a cddl type for that (`time`) that maps to CBOR type  `#6.1`, see:
* https://www.rfc-editor.org/rfc/rfc8610.html#appendix-D
*  https://www.rfc-editor.org/rfc/rfc8949.html#name-epoch-based-date-time

> TimeLimited | `register_by: uint` | Returns true if now () < `register_by` at registration time. The ledger MUST store the ledger time at registration along with the claim, and SHOULD indicate it in receipts. The value provided for `register_by` MUST be a Unix timestamp, i.e., a `uint` representing the number of seconds since 00:00:00 UTC, 1 January 1970, without leap seconds.

Reviewer 3: "without leap seconds" is ambiguous.
Please refer to Section 3.4.2 of RFC 8949 for a precise definition; of course, you would limit this to unsigned integers (which is analog to the last sentence of the penultimate paragraph of that Section).
(See [my other comment](https://github.com/ietf-scitt/draft-birkholz-scitt-architecture/pull/41#issuecomment-1351160877) for how to make this evident in the CDDL too, based on Henk's comment.)

Author: The CDDL is now meaningful, reflecting your other comments. That makes this explanation redundant, however I'm inclined to keep a brief explanation to aid most readers, so I've updated the wording to reflect Section 3.4.2 of RFC 8949 as you suggest.

Sequential | `sequence_no: uint` | First, lookup in the ledger for claims with the same issuer and feed. If at least one is found, returns true if and only if the `sequence_no` of the new claim is the highest `sequence_no` in the existing claims incremented by one. Otherwise, returns true if and only if `sequence_no = 0`.

> Temporal | `issuance_ts: uint` | Returns true if and only if there is no claim in the ledger with the same issuer and feed with a greater `issuance_ts`. The value provided for `issuance_ts` MUST be a Unix timestamp e.g. a `uint` representing the number of seconds since 00:00:00 UTC, 1 January 1970, minus leap seconds.

Reviewer: ```suggestion
Temporal | `issuance_ts: uint` | Returns true if and only if there is no claim in the ledger with the same issuer and feed with a greater `issuance_ts`. The value provided for `issuance_ts` MUST be a Unix timestamp, i.e., a `uint` representing the number of seconds since 00:00:00 UTC, 1 January 1970, without leap seconds.
```

> Temporal | `issuance_ts: uint` | Returns true if and only if there is no claim in the ledger with the same issuer and feed with a greater `issuance_ts` and now () > `issuance_ts` at registration time. The value provided for `issuance_ts` MUST be a Unix timestamp, i.e., a `uint` representing the number of seconds since 00:00:00 UTC, 1 January 1970, without leap seconds.

Reviewer 3: I'm looking closer for the first time -- is there no need for a reg-info to be compact?
(You are using text string labels, which are less compact than integer labels would be, which is OK if they don't have to be compact.)

Author: This may be desirable, but I call it out of scope for this PR. I've opened an issue to track it: https://github.com/ietf-scitt/draft-birkholz-scitt-architecture/issues/43

> NoReplay | `no_replay: null` | If the `no_replay` attribute is present then the policy returns true if and only if the claim doesn't already appear in the ledger.

Reviewer 3: more idiomatic would be `? no_replay: true`.
(Of course, a "Unit" [single-element] type could be created from a tag, but that is unnecessary here, I think.)

Author: Viewing reg-info as mapping between policies and their required attributes, then `null` makes more sense as `no_replay` has no required attributes. 
(I also worry that using `true` will invite the mistake of trying to set `no_replay` to `false`.)

Author: In the not too distant future it might make sense to change to structure of reg-info to make it more clearly a mapping between policies and their required attributes or some similar structure, perhaps motivated by policies with multiple required attributes - but I don't think it's worth doing that in this PR.
